### PR TITLE
Fixed #1088

### LINKED ISSUE #1088: TCK failures : 2 SeBootstrapIT tests fail in Linux & mac due to permission issue
2 below SeBootstrapIT tests **fail when run in Linux and Mac** for Jersey(M2)

- ee.jakarta.tck.ws.rs.sebootstrap.SeBootstrapIT.shouldBootInstanceUsingImplementationsDefaultIpPort
- ee.jakarta.tck.ws.rs.sebootstrap.SeBootstrapIT.shouldBootInstanceUsingDefaults

with the below errors.

```
[ERROR] ee.jakarta.tck.ws.rs.sebootstrap.SeBootstrapIT.shouldBootInstanceUsingImplementationsDefaultIpPort  Time elapsed: 0.055 s  <<< ERROR!
java.util.concurrent.ExecutionException: jakarta.ws.rs.ProcessingException: Failed to start Grizzly HTTP server: Permission denied
        at ee.jakarta.tck.ws.rs.sebootstrap.SeBootstrapIT.shouldBootInstanceUsingImplementationsDefaultIpPort(SeBootstrapIT.java:314)
Caused by: jakarta.ws.rs.ProcessingException: Failed to start Grizzly HTTP server: Permission denied
Caused by: **java.net.SocketException: Permission denied**

[ERROR] ee.jakarta.tck.ws.rs.sebootstrap.SeBootstrapIT.shouldBootInstanceUsingDefaults  Time elapsed: 0.042 s  <<< ERROR!
java.util.concurrent.ExecutionException: jakarta.ws.rs.ProcessingException: Failed to start Grizzly HTTP server: Permission denied
        at ee.jakarta.tck.ws.rs.sebootstrap.SeBootstrapIT.shouldBootInstanceUsingDefaults(SeBootstrapIT.java:73)
Caused by: jakarta.ws.rs.ProcessingException: Failed to start Grizzly HTTP server: Permission denied
Caused by: **java.net.SocketException: Permission denied**
```

**The tests were reported as passing in Windows.**

The issue was fixed by running `sudo /sbin/sysctl -w net.ipv4.ip_unprivileged_port_start=0` in linux machine to allow non root users to have access to all ports. Is there any change required in the test code to get the test pass without the above workaround? Should we be using a **port number >1024 to run these 2 tests**? 

@mkarg @jansupol @spericas

### PR DESCRIPTION
**Pull request by mkarg**
For fixing #1088 this PR introduces a new, unit-tested utility method in the TCK for obtaining an unprivileged IP port.

**As these are no API changes, this is a fast-track review period of just one day as per our [committer rules](https://github.com/eclipse-ee4j/jaxrs-api/wiki/Committer-Conventions#minimum-length-of-review-period-before-merge--close-of-prs-and-closing-of-issues).**

### CODE REVIEW

File: jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/util/JaxrsUtil.java [hunk 1]
```diff
@@ -34,6 +35,16 @@
 
 public abstract class JaxrsUtil {
 
+  public static final int unprivilegedPort() throws IOException {
+    int port = 0;
+    while (port < 1024) {
+      try (final ServerSocket serverSocket = new ServerSocket(0)) {
+        port = serverSocket.getLocalPort();
+      }
+    }
```
Comment (spericas) [inline jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/util/JaxrsUtil.java:44]: This loop has the potential of being infinite, in theory. We should cap the number retries.

Comment (mkarg) [reply jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/util/JaxrsUtil.java:44]: I doubt that you will be able to actually detect that endless loop in reality (can you proof your claim?), but I can add a cap at 1025 tries (1 more than the size of the privileged range).

Comment (spericas) [reply jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/util/JaxrsUtil.java:44]: You can't easily prove it either way, that's the point.

**Review by spericas — DISMISSED** [review]

**Review by andymc12 — DISMISSED** [review]
LGTM

**Comment by mkarg** [comment]
As the package was renamed meanwhile, I had to push another commit into the PR. As a consequence, I need to ask everybody to re-cast his vote. Thanks.

**Review by spericas — APPROVED** [review]

**Review by jansupol — APPROVED** [review]

**Pull request merged**